c10/util/Exception.h

```diff
@@ -56,13 +66,29 @@ class C10_API Error : public std::exception {
       const std::string& backtrace,
       const void* caller = nullptr);
 
-  void AppendMessage(const std::string& msg);
+  // Base constructor
+  Error(
+      const std::string& msg,
+      const std::string& backtrace,
+      const void* caller = nullptr);
+
+  // Add some new context to the message stack (pushing it to the top
+  // of the message stack)
```

Comment:
Nit: "top" might benefit from some qualification here, since it's actually going to go to the end of the formatted message IIUC